Update Chapter 9

danielvdende opened the pull request.

krisgeus commented on chapter09/README.md:
````diff
@@ -0,0 +1,43 @@
+During the work for this chapter, a bug in Airflow forced a downgrade of the Postgres provider. The bug is described [here](https://github.com/apache/airflow/issues/41373).
+The workaround was to pin the Postgres provider to version 5.0.0:
+```
+apache-airflow-providers-postgres==5.0.0
````
This i not reflected in the .env file as _PIP_ADDITIONAL_REQUIREMENTS

krisgeus commented on chapter09/README.md:
````diff
@@ -0,0 +1,43 @@
+To get started with the code examples, start Airflow in docker using the following command:
+
+```
+docker-compose up -d --build
````
`--build` not needed since we do not have custom docker builds referenced in the compose.override.yaml

krisgeus commented on chapter09/compose.yaml:
```diff
@@ -0,0 +1,288 @@
+# Licensed to the Apache Software Foundation (ASF) under one
```
This seems to be a copy and needs to be a symbolic ling to the file in the parent directory

danielvdende replied:
I don't quite understand what you mean?

krisgeus commented on chapter09/custom/movielens_download_operator.py:
```diff
@@ -0,0 +1,33 @@
+
+from airflow.models import BaseOperator
+
+from chapter09.custom.movielens_hook import MovielensHook
```
There is no pip install of the custom components anywhere. How does this work?

danielvdende replied:
Fixed

krisgeus commented on chapter09/dags/03_duplicate_task_ids.py:
```diff
@@ -0,0 +1,13 @@
+from airflow.operators.empty import EmptyOperator
+
+dag = DAG(
+    dag_id="chapter8_duplicate_task_ids",
```
Dag Id says chapter 8 instead of 03

krisgeus commented on chapter09/dags/03_duplicate_task_ids.py:
```diff
@@ -0,0 +1,13 @@
+import pendulum
```
This dag does not show up in the UI with the error. Is that correct?

danielvdende replied:
Yes. I've now added it to `.airflowignore` to avoid this error. These files are intended for the dag_integrity test

krisgeus commented on chapter09/dags/01_dag_cycle.py:
```diff
@@ -0,0 +1,15 @@
+import pendulum
```
This dag does not show up in the UI with the error. Is that correct?

danielvdende replied:
Yes. I've now added it to `.airflowignore` to avoid this error. These files are intended for the dag_integrity test

krisgeus commented on chapter09/dags/02_bash_operator_no_command.py:
```diff
@@ -2,10 +2,9 @@
 
-dag = DAG(
+with DAG(
     dag_id="chapter8_bash_operator_no_command",
```
Dag is says chapter 8

danielvdende replied:
Fixed

krisgeus commented on chapter09/dags/dagtestdag.py:
```diff
@@ -0,0 +1,27 @@
+
+from airflow import DAG
+
+from chapter09.custom.movielens_to_postgres_operator import MovielensToPostgresOperator
```
Custom stuff not installed so this dag gives error

danielvdende replied:
Fixed

danielvdende commented:
@krisgeus Thanks for your first review! I've reworked a couple of things:
1. The folder structure is now in line with that of chapter 8, which resolves the issues with rendering the dags with the custom hook/operator in Airflow.
2. I've added instructions on how to run the tests. While writing the chapter, I actually didn't really use the docker-compose environment at all, as I was mostly working in my IDE. I considered adding a separate pytest container to docker-compose, but that could get a bit complicated as some of the tests use Docker containers themselves for spinning up test-related services (i.e. the postgres ones). I figured it wouldn't be the best use of my time to figure that out. Instead, I've listed the requirements with some instructions on how to use in the README.

krisgeus commented on chapter09/README.md:
````diff
@@ -0,0 +1,58 @@
+
+You should now be able to run the tests, e.g.
+```bash
+pytest tests/
````
We need to execute `airflow db init` to instantiate an airflow DB

krisgeus reviewed (approved):
LGTM

krisgeus commented on chapter09/tests/dags/01_test_dag_integrity.py:
```diff
@@ -0,0 +1,24 @@
+"""Test integrity of DAGs."""
```
This test is not automaticcaly resolved by pytest. You need to name it `test_something.py`

krisgeus commented on chapter09/tests/dags/02_test_example.py:
```diff
@@ -0,0 +1,9 @@
+from airflow.operators.bash import BashOperator
```
This test is not automatically resolved by pytest. You need to name it test_something.py

krisgeus commented on chapter09/tests/dags/01_test_dag_integrity.py:
```diff
@@ -0,0 +1,24 @@
+from airflow.dag_processing.processor import DagFileProcessor
+
+DAG_PATH = os.path.join(
+    os.path.dirname(__file__), "..", "..", "dags/02_bash_operator_no_command.py"
```
Only 02 dag is integrity tested because of this. Maybe change to `"dags/*.py"`

krisgeus commented on chapter09/tests/conftest.py:
```diff
@@ -1,6 +1,7 @@
 
 import pytest
+import uuid
 from airflow.models import DAG, BaseOperator
```
BaseOperator not used ??
